src/lib.rs

```diff
@@ -120,6 +126,31 @@ impl Module {
         unsafe { ffi::BinaryenModuleOptimize(self.inner.raw) }
     }
 
+    /// Run a specified set of optimization passes on the module.
+    ///
+    /// This WILL NOT take into account code generation configuration set by `set_global_codegen_config`.
+    pub fn run_optimization_passes(&self, passes: &Vec<String>) -> Result<(), ()> {
```

Comment:
This code can be a little bit more friendlier to it's users. It asks users for an allocated `Vec` of allocated `String`s. However, it doesn't actually require ownership on those and could get away with a slice instead of `Vec` and `&str` instead of an owned `String`. 

We could use `&[&str]` for passes, but that might be also not super convenient if user does have owned `Vec<String>`. So I think something like this should work:


```rust
    /// Run a specified set of optimization passes on the module.
    ///
    /// This WILL NOT take into account code generation configuration set by `set_global_codegen_config`.
    pub fn run_optimization_passes<B: AsRef<str>, I: IntoIterator<Item = B>>(
        &self,
        passes: I,
    ) -> Result<(), ()> {
        let mut cstr_vec: Vec<_> = vec![];
        for pass in passes {
            if !is_valid_pass(pass.as_ref()) {
                return Err(());
            }

            cstr_vec.push(CString::new(pass.as_ref()).unwrap());
        }

        // NOTE: BinaryenModuleRunPasses expectes a mutable ptr
        let mut ptr_vec: Vec<_> = cstr_vec.iter().map(|pass| pass.as_ptr()).collect();

        unsafe {
            ffi::BinaryenModuleRunPasses(self.inner.raw, ptr_vec.as_mut_ptr(), ptr_vec.len() as u32)
        };
        Ok(())
    }
```

This code should work both with owned (i.e. `run_optimization_passes(&vec!["invalid".to_string()])`) and borrowed structures (i.e. `run_optimization_passes(&["invalid"])`)